crates/wallet/src/wallet/coin_selection.rs

```diff
@@ -297,15 +297,15 @@ impl CoinSelectionAlgorithm for OldestFirstCoinSelection {
 /// - `remaining_amount`: the amount in which the selected coins exceed the target amount
 /// - `fee_rate`: required fee rate for the current selection
 /// - `drain_script`: script to consider change creation
-pub fn decide_change(remaining_amount: u64, fee_rate: FeeRate, drain_script: &Script) -> Excess {
+pub fn decide_change(remaining_amount: Amount, fee_rate: FeeRate, drain_script: &Script) -> Excess {
     // drain_output_len = size(len(script_pubkey)) + len(script_pubkey) + size(output_value)
     let drain_output_len = serialize(drain_script).len() + 8usize;
     let change_fee =
-        (fee_rate * Weight::from_vb(drain_output_len as u64).expect("overflow occurred")).to_sat();
-    let drain_val = remaining_amount.saturating_sub(change_fee);
+        fee_rate * Weight::from_vb(drain_output_len as u64).expect("overflow occurred");
+    let drain_val = remaining_amount.checked_sub(change_fee).unwrap_or_default();
```

Comment:
Is `Default` here `Amount::ZERO`?

Reply:
Yup default for `Amount` is `Amount::ZERO`.